StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -49,158 +51,161 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+                    var partOfLambdaExpression = token.Parent.Parent.IsKind(SyntaxKind.ParenthesizedLambdaExpression);
+                    haveLeadingSpace = partOfLambdaExpression;
+                    haveTrailingSpace = false;
+                    break;
 
                 default:
-                    if (precedingToken.Parent is BinaryExpressionSyntax
-                        || precedingToken.Parent is AssignmentExpressionSyntax
-                        || precedingToken.Parent is ConditionalExpressionSyntax
-                        || precedingToken.Parent is EqualsValueClauseSyntax)
-                    {
-                        allowLeadingNoSpace = false;
-                        allowLeadingSpace = true;
-                        break;
-                    }
-                    else
-                    {
-                        allowLeadingNoSpace = true;
-                        allowLeadingSpace = false;
-                        break;
-                    }
-                }
+                    haveLeadingSpace = false;
+                    haveTrailingSpace = false;
+                    break;
             }
 
-            followedBySpace = token.HasTrailingTrivia;
-            lastInLine = followedBySpace && token.TrailingTrivia.Any(SyntaxKind.EndOfLineTrivia);
-
-            if (reportPrecedingError && !firstInLine)
+            // Ignore spacing before if another opening parenthesis is before this.
+            // That way the first opening parenthesis will report any spacing errors.
+            if (!prevTokenIsOpenParen)
             {
-                if (!allowLeadingSpace && precededBySpace)
-                {
-                    // Opening parenthesis must{ not} be {preceded} by a space.
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, token.GetLocation(), " not", "preceded"));
-                }
-                else if (!allowLeadingNoSpace && !precededBySpace)
+                var hasLeadingComment = (leadingTriviaList.Count > 0) && leadingTriviaList.Last().IsKind(SyntaxKind.MultiLineCommentTrivia);
+                var hasLeadingSpace = (leadingTriviaList.Count > 0) && leadingTriviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+
+                if (!isFirstOnLine && !hasLeadingComment && (haveLeadingSpace ^ hasLeadingSpace))
                 {
-                    // Opening parenthesis must{} be {preceded} by a space.
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, token.GetLocation(), string.Empty, "preceded"));
+                    var properties = new Dictionary<string, string>
+                    {
+                        { "location", "preceding" },
+                        { "action", haveLeadingSpace ? "insert" : "remove" }
+                    };
+
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, token.GetLocation(), properties.ToImmutableDictionary(), haveLeadingSpace ? string.Empty : " not", "preceded"));
```

Comment:
:bulb: Consider using multiple `DiagnosticDescriptor` instances (similar to SA1612). The current format can't be easily localized, but if we use multiple descriptors they can each be easily translated to other languages.